plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/BaseDeltaLakeConnectorSmokeTest.java

```diff
@@ -1746,6 +1750,34 @@ public void testVacuum()
             assertThat(getActiveFiles(tableName)).isEqualTo(updatedFiles);
             // old files should be cleaned up
             assertThat(getAllDataFilesFromTableDirectory(tableName)).isEqualTo(updatedFiles);
+            // operations should be logged
+            assertThat(query("SELECT version, operation, MAP_FILTER(operation_parameters, (k, v) -> k <> 'queryId') FROM \"" + tableName + "$history\"")).matches( """
+                    VALUES\s
+                    (CAST(0 AS BIGINT), CAST('CREATE TABLE AS SELECT' AS VARCHAR), CAST(MAP() AS MAP(VARCHAR, VARCHAR))),\s
+                    (1, 'MERGE', MAP()),\s
+                    (2, 'VACUUM START', MAP()),\s
+                    (3, 'VACUUM END', MAP(ARRAY['status'], ARRAY['COMPLETED'])),\s
+                    (4, 'VACUUM START', MAP()),\s
+                    (5, 'VACUUM END', MAP(ARRAY['status'], ARRAY['COMPLETED']))""");
+            assertThat(Integer.parseInt((String) computeScalar("SELECT operation_metrics['sizeOfDataToDelete'] FROM \"" + tableName + "$history\" WHERE version = 2"))).isEqualTo(0);
+            assertThat(Integer.parseInt((String) computeScalar("SELECT operation_metrics['numFilesToDelete'] FROM \"" + tableName + "$history\" WHERE version = 2"))).isEqualTo(0);
+            assertThat(Integer.parseInt((String) computeScalar("SELECT operation_metrics['numDeletedFiles'] FROM \"" + tableName + "$history\" WHERE version = 3"))).isEqualTo(0);
+            assertThat(Integer.parseInt((String) computeScalar("SELECT operation_metrics['numVacuumedDirectories'] FROM \"" + tableName + "$history\" WHERE version = 3"))).isEqualTo(8);
+            assertThat(Integer.parseInt((String) computeScalar("SELECT operation_metrics['sizeOfDataToDelete'] FROM \"" + tableName + "$history\" WHERE version = 4"))).isGreaterThanOrEqualTo(5012);
+            assertThat(Integer.parseInt((String) computeScalar("SELECT operation_metrics['numFilesToDelete'] FROM \"" + tableName + "$history\" WHERE version = 4"))).isEqualTo(5);
+            assertThat(Integer.parseInt((String) computeScalar("SELECT operation_metrics['numDeletedFiles'] FROM \"" + tableName + "$history\" WHERE version = 5"))).isEqualTo(5);
+            assertThat(Integer.parseInt((String) computeScalar("SELECT operation_metrics['numVacuumedDirectories'] FROM \"" + tableName + "$history\" WHERE version = 5"))).isEqualTo(8);
+
+            assertUpdate(sessionWithVacuumLoggingDisabled, "CALL system.vacuum(schema_name => CURRENT_SCHEMA, table_name => '" + tableName + "', retention => '1s')");
+            // no new vacuum logging operations are logged
+            assertQuery("SELECT version, operation FROM \"" + tableName + "$history\"", """
+                    VALUES\s
+                    (0, 'CREATE TABLE AS SELECT'),\s
+                    (1, 'MERGE'),\s
+                    (2, 'VACUUM START'),\s
+                    (3, 'VACUUM END'),\s
+                    (4, 'VACUUM START'),\s
+                    (5, 'VACUUM END')""");
```

Comment:
Can we add another test method that writes Parquet checkpoint in VACUUM?  You can use `checkpoint_interval` table property. 

As you may know, the connector writes two formats (JSON, Parquet) as transaction logs. We want to cover both formats.

Reply:
Added the test, but there is some code duplication now.